C++ demos don't compile

I didn't propagate the ThreadSchedule changes into the C++ demos. :( b/c I only built the library in testing.

Resulting in errors like:

```
/Users/eliu/src/MOE/moe/optimal_learning/cpp/gpp_expected_improvement_demo.cpp: In function 'int main()':
/Users/eliu/src/MOE/moe/optimal_learning/cpp/gpp_expected_improvement_demo.cpp:220:259: error: no matching function for call to 'ComputeOptimalPointsToSampleWithRandomStarts(optimal_learning::GaussianProcess&, optimal_learning::GradientDescentParameters&, DomainType&, double*, const int&, const int&, double&, int&, int&, bool*, optimal_learning::UniformRandomGenerator*, optimal_learning::NormalRNG*, double*)'
/Users/eliu/src/MOE/moe/optimal_learning/cpp/gpp_expected_improvement_demo.cpp:220:259: note: candidate is:
In file included from /Users/eliu/src/MOE/moe/optimal_learning/cpp/gpp_expected_improvement_demo.cpp:45:0:
/Users/eliu/src/MOE/moe/optimal_learning/cpp/gpp_math.hpp:1447:6: note: template<class DomainType> void optimal_learning::ComputeOptimalPointsToSampleWithRandomStarts(const optimal_learning::GaussianProcess&, const optimal_learning::GradientDescentParameters&, const DomainType&, const optimal_learning::ThreadSchedule&, const double*, int, int, double, int, bool*, optimal_learning::UniformRandomGenerator*, optimal_learning::NormalRNG*, double*)
/Users/eliu/src/MOE/moe/optimal_learning/cpp/gpp_math.hpp:1447:6: note:   template argument deduction/substitution failed:
/Users/eliu/src/MOE/moe/optimal_learning/cpp/gpp_expected_improvement_demo.cpp:220:259: note:   cannot convert 'points_being_sampled.std::vector<_Tp, _Alloc>::data<double, std::allocator<double> >()' (type 'double*') to type 'const optimal_learning::ThreadSchedule&'
make[2]: *** [CMakeFiles/gpp_hyper_and_EI_demo.dir/gpp_hyper_and_EI_demo.cpp.o] Error 1
```

--- a/moe/optimal_learning/cpp/gpp_expected_improvement_demo.cpp
+++ b/moe/optimal_learning/cpp/gpp_expected_improvement_demo.cpp
@@ -43,6 +43,7 @@
 #include "gpp_domain.hpp"
 #include "gpp_logging.hpp"
 #include "gpp_math.hpp"
+#include "gpp_optimization.hpp"
 #include "gpp_optimization_parameters.hpp"
 #include "gpp_random.hpp"
 #include "gpp_test_utils.hpp"
@@ -79,6 +80,7 @@ int main() {
 
   // multithreading
   int max_num_threads = 1;  // feel free to experiment with different numbers
+  ThreadSchedule thread_schedule(max_num_threads, omp_sched_dynamic);
 
   // set up RNG containers
   int64_t pi_array[] = {314, 3141, 31415, 314159, 3141592, 31415926, 314159265, 3141592653, 31415926535, 314159265359};  // arbitrarily used digits of pi as seeds
@@ -148,9 +150,11 @@ int main() {
   domain.GenerateUniformPointsInDomain(num_sampled, &uniform_generator, points_sampled.data());
 
   // build an empty GP: since num_sampled (last arg) is 0, none of the data arrays will be used here
-  GaussianProcess gp_generator(covariance_original, points_sampled.data(), points_sampled_value.data(), noise_variance.data(), dim, 0);
+  GaussianProcess gp_generator(covariance_original, points_sampled.data(), points_sampled_value.data(),
+                               noise_variance.data(), dim, 0);
   // fill the GP with randomly generated data
-  FillRandomGaussianProcess(points_sampled.data(), noise_variance.data(), dim, num_sampled, points_sampled_value.data(), &gp_generator);
+  FillRandomGaussianProcess(points_sampled.data(), noise_variance.data(), dim, num_sampled,
+                            points_sampled_value.data(), &gp_generator);
 
   // set points_being_sampled
   // doing this arbitrarily here, but you could imagine running EI opt once with 0 points_being_sampled,
@@ -173,12 +177,14 @@ int main() {
   std::vector<double> hyperparameters_perturbed(covariance_original.GetNumberOfHyperparameters());
   boost::uniform_real<double> uniform_double_for_wrong_hyperparameter(0.01, 5.0);
   CovarianceClass covariance_perturbed(dim, 1.0, 1.0);
-  FillRandomCovarianceHyperparameters(uniform_double_for_wrong_hyperparameter, &uniform_generator, &hyperparameters_perturbed, &covariance_perturbed);
+  FillRandomCovarianceHyperparameters(uniform_double_for_wrong_hyperparameter, &uniform_generator,
+                                      &hyperparameters_perturbed, &covariance_perturbed);
 
   // Note: with random data generation, technically we already have the GP ready since
   // we have been progressively adding new points to it.  Still we will construct a new GP
   // for the purpose of demonstration
-  GaussianProcess gp(covariance_perturbed, points_sampled.data(), points_sampled_value.data(), noise_variance.data(), dim, num_sampled);
+  GaussianProcess gp(covariance_perturbed, points_sampled.data(), points_sampled_value.data(),
+                     noise_variance.data(), dim, num_sampled);
 
   // remaining inputs to EI optimization
   double best_so_far = *std::min_element(points_sampled_value.begin(), points_sampled_value.end());  // this is simply the best function value seen to date
@@ -191,7 +197,8 @@ int main() {
   int num_multistarts = 30;  // max number of multistarted locations
   int max_num_steps = 500;  // maximum number of GD iterations per restart
   int max_num_restarts = 20;  // number of restarts to run with GD
-  GradientDescentParameters gd_params(num_multistarts, max_num_steps, max_num_restarts, gamma, pre_mult, max_relative_change, tolerance);
+  GradientDescentParameters gd_params(num_multistarts, max_num_steps, max_num_restarts, gamma,
+                                      pre_mult, max_relative_change, tolerance);
   // so the total number of GD iterations is at most:
   // num_multistarts * max_num_restarts * max_num_steps
 
@@ -217,7 +224,11 @@ int main() {
   printf(OL_ANSI_COLOR_CYAN "OPTIMIZING EXPECTED IMPROVEMENT...\n" OL_ANSI_COLOR_RESET);
   std::vector<double> next_point_winner(dim);
   bool found_flag = false;
-  ComputeOptimalPointsToSampleWithRandomStarts(gp, gd_params, domain, points_being_sampled.data(), num_to_sample, num_being_sampled, best_so_far, max_int_steps, max_num_threads, &found_flag, &uniform_generator, normal_rng_vec.data(), next_point_winner.data());
+  ComputeOptimalPointsToSampleWithRandomStarts(gp, gd_params, domain, thread_schedule,
+                                               points_being_sampled.data(), num_to_sample,
+                                               num_being_sampled, best_so_far, max_int_steps,
+                                               &found_flag, &uniform_generator, normal_rng_vec.data(),
+                                               next_point_winner.data());
   printf(OL_ANSI_COLOR_CYAN "EI OPTIMIZATION FINISHED. Success status: %s\n" OL_ANSI_COLOR_RESET, found_flag ? "True" : "False");
   printf("Next best sample point according to EI:\n");
   PrintMatrix(next_point_winner.data(), 1, dim);

--- a/moe/optimal_learning/cpp/gpp_hyper_and_EI_demo.cpp
+++ b/moe/optimal_learning/cpp/gpp_hyper_and_EI_demo.cpp
@@ -45,6 +45,7 @@
 #include "gpp_logging.hpp"
 #include "gpp_math.hpp"
 #include "gpp_model_selection.hpp"
+#include "gpp_optimization.hpp"
 #include "gpp_optimization_parameters.hpp"
 #include "gpp_random.hpp"
 #include "gpp_test_utils.hpp"
@@ -95,7 +96,8 @@ int main() {
   // generate randomly
   boost::uniform_real<double> uniform_double_for_hyperparameter(0.5, 1.5);
   CovarianceClass covariance_original(dim, 1.0, 1.0);
-  FillRandomCovarianceHyperparameters(uniform_double_for_hyperparameter, &uniform_generator, &hyperparameters_original, &covariance_original);
+  FillRandomCovarianceHyperparameters(uniform_double_for_hyperparameter, &uniform_generator,
+                                      &hyperparameters_original, &covariance_original);
 
   // Generate data that will be used to build the GP
   // set noise
@@ -105,24 +107,29 @@ int main() {
   domain.GenerateUniformPointsInDomain(num_sampled, &uniform_generator, points_sampled.data());
 
   // build an empty GP: since num_sampled (last arg) is 0, none of the data arrays will be used here
-  GaussianProcess gp_generator(covariance_original, points_sampled.data(), points_sampled_value.data(), noise_variance.data(), dim, 0);
+  GaussianProcess gp_generator(covariance_original, points_sampled.data(), points_sampled_value.data(),
+                               noise_variance.data(), dim, 0);
   // fill the GP with randomly generated data
-  FillRandomGaussianProcess(points_sampled.data(), noise_variance.data(), dim, num_sampled, points_sampled_value.data(), &gp_generator);
+  FillRandomGaussianProcess(points_sampled.data(), noise_variance.data(), dim, num_sampled,
+                            points_sampled_value.data(), &gp_generator);
 
   // simulate not knowing the hyperparameters by choosing a new set randomly (start hyperparameter opt from here)
   std::vector<double> hyperparameters_perturbed(covariance_original.GetNumberOfHyperparameters());
   // choose from a small-ish range; for the demo we want hyperparameter opt to converge fairly quickly/robustly
   boost::uniform_real<double> uniform_double_for_perturbed_hyperparameter(4.0, 8.0);
   CovarianceClass covariance_perturbed(dim, 1.0, 1.0);
-  FillRandomCovarianceHyperparameters(uniform_double_for_perturbed_hyperparameter, &uniform_generator, &hyperparameters_perturbed, &covariance_perturbed);
+  FillRandomCovarianceHyperparameters(uniform_double_for_perturbed_hyperparameter, &uniform_generator,
+                                      &hyperparameters_perturbed, &covariance_perturbed);
 
   std::vector<ClosedInterval> hyperparameter_domain_bounds(covariance_original.GetNumberOfHyperparameters(), {1.0e-10, 1.0e10});
-  HyperparameterDomainType hyperparameter_domain(hyperparameter_domain_bounds.data(), covariance_original.GetNumberOfHyperparameters());
+  HyperparameterDomainType hyperparameter_domain(hyperparameter_domain_bounds.data(),
+                                                 covariance_original.GetNumberOfHyperparameters());
 
   // Log Likelihood eval
   using LogLikelihoodEvaluator = LogMarginalLikelihoodEvaluator;
   // log likelihood evaluator object
-  LogLikelihoodEvaluator log_marginal_eval(points_sampled.data(), points_sampled_value.data(), noise_variance.data(), dim, num_sampled);
+  LogLikelihoodEvaluator log_marginal_eval(points_sampled.data(), points_sampled_value.data(),
+                                           noise_variance.data(), dim, num_sampled);
 
   // Newton setup
   int total_newton_errors = 0;  // number of newton runs that failed due to singular hessians
@@ -131,7 +138,8 @@ int main() {
   double pre_mult_newton = 1.0e-1;  // newton diagonal dominance scaling factor (see newton docs for details)
   double max_relative_change_newton = 1.0;
   double tolerance_newton = 1.0e-11;
-  NewtonParameters newton_parameters(1, newton_max_num_steps, gamma_newton, pre_mult_newton, max_relative_change_newton, tolerance_newton);
+  NewtonParameters newton_parameters(1, newton_max_num_steps, gamma_newton, pre_mult_newton,
+                                     max_relative_change_newton, tolerance_newton);
 
   std::vector<double> new_newton_hyperparameters(covariance_original.GetNumberOfHyperparameters());
 
@@ -141,7 +149,9 @@ int main() {
 
   printf(OL_ANSI_COLOR_CYAN "NEWTON OPTIMIZED HYPERPARAMETERS:\n" OL_ANSI_COLOR_RESET);
   // run newton optimization
-  total_newton_errors += NewtonHyperparameterOptimization(log_marginal_eval, covariance_perturbed, newton_parameters, hyperparameter_domain, new_newton_hyperparameters.data());
+  total_newton_errors += NewtonHyperparameterOptimization(log_marginal_eval, covariance_perturbed,
+                                                          newton_parameters, hyperparameter_domain,
+                                                          new_newton_hyperparameters.data());
 
   printf("Result of newton:\n");
   PrintMatrix(new_newton_hyperparameters.data(), 1, covariance_original.GetNumberOfHyperparameters());
@@ -153,7 +163,8 @@ int main() {
   // Now optimize EI using the 'best' hyperparameters
   // set gaussian process's hyperparameters to the result of newton optimization
   CovarianceClass covariance_opt(dim, new_newton_hyperparameters[0], new_newton_hyperparameters.data() + 1);
-  GaussianProcess gp_model(covariance_opt, points_sampled.data(), points_sampled_value.data(), noise_variance.data(), dim, num_sampled);
+  GaussianProcess gp_model(covariance_opt, points_sampled.data(), points_sampled_value.data(),
+                           noise_variance.data(), dim, num_sampled);
 
   // remaining inputs to EI optimization
   // just an arbitrary point set for when num_being_sampled = 2, as in the default setting for this demo
@@ -165,6 +176,7 @@ int main() {
 
   // multithreading
   int max_num_threads = 2;  // feel free to experiment with different numbers
+  ThreadSchedule thread_schedule(max_num_threads, omp_sched_dynamic);
 
   // set up RNG containers
   int64_t pi_array[] = {314, 3141, 31415, 314159, 3141592, 31415926, 314159265, 3141592653, 31415926535, 314159265359};  // arbitrarily used digits of pi as seeds
@@ -184,7 +196,8 @@ int main() {
   int num_multistarts = 10;  // max number of multistarted locations
   int max_num_steps = 500;  // maximum number of GD iterations per restart
   int max_num_restarts = 20;  // number of restarts to run with GD
-  GradientDescentParameters gd_params(num_multistarts, max_num_steps, max_num_restarts, gamma, pre_mult, max_relative_change_ei, tolerance_ei);
+  GradientDescentParameters gd_params(num_multistarts, max_num_steps, max_num_restarts, gamma,
+                                      pre_mult, max_relative_change_ei, tolerance_ei);
 
   // EI evaluation parameters
   int max_int_steps = 1000;  // number of monte carlo iterations
@@ -193,7 +206,11 @@ int main() {
   {  // optimize EI using a model with the optimized hyperparameters
     printf(OL_ANSI_COLOR_CYAN "OPTIMIZING EXPECTED IMPROVEMENT... (optimized hyperparameters)\n" OL_ANSI_COLOR_RESET);
     bool found_flag = false;
-    ComputeOptimalPointsToSampleWithRandomStarts(gp_model, gd_params, domain, points_being_sampled.data(), num_to_sample, num_being_sampled, best_so_far, max_int_steps, max_num_threads, &found_flag, &uniform_generator, normal_rng_vec.data(), next_point_winner.data());
+    ComputeOptimalPointsToSampleWithRandomStarts(gp_model, gd_params, domain, thread_schedule,
+                                                 points_being_sampled.data(), num_to_sample,
+                                                 num_being_sampled, best_so_far, max_int_steps,
+                                                 &found_flag, &uniform_generator, normal_rng_vec.data(),
+                                                 next_point_winner.data());
     printf(OL_ANSI_COLOR_CYAN "EI OPTIMIZATION FINISHED (optimized hyperparameters). Success status: %s\n" OL_ANSI_COLOR_RESET, found_flag ? "True" : "False");
     printf("Next best sample point according to EI (opt hyper):\n");
     PrintMatrix(next_point_winner.data(), 1, dim);
@@ -215,11 +232,17 @@ int main() {
     std::vector<double> hyperparameters_wrong(covariance_original.GetNumberOfHyperparameters());
     boost::uniform_real<double> uniform_double_for_wrong_hyperparameter(0.1, 0.5);
     CovarianceClass covariance_wrong(dim, 1.0, 1.0);
-    FillRandomCovarianceHyperparameters(uniform_double_for_wrong_hyperparameter, &uniform_generator, &hyperparameters_wrong, &covariance_wrong);
-    GaussianProcess gp_wrong_hyper(covariance_wrong, points_sampled.data(), points_sampled_value.data(), noise_variance.data(), dim, num_sampled);
+    FillRandomCovarianceHyperparameters(uniform_double_for_wrong_hyperparameter, &uniform_generator,
+                                        &hyperparameters_wrong, &covariance_wrong);
+    GaussianProcess gp_wrong_hyper(covariance_wrong, points_sampled.data(), points_sampled_value.data(),
+                                   noise_variance.data(), dim, num_sampled);
 
     bool found_flag = false;
-    ComputeOptimalPointsToSampleWithRandomStarts(gp_wrong_hyper, gd_params, domain, points_being_sampled.data(), num_to_sample, num_being_sampled, best_so_far, max_int_steps, max_num_threads, &found_flag, &uniform_generator, normal_rng_vec.data(), next_point_winner.data());
+    ComputeOptimalPointsToSampleWithRandomStarts(gp_wrong_hyper, gd_params, domain, thread_schedule,
+                                                 points_being_sampled.data(), num_to_sample,
+                                                 num_being_sampled, best_so_far, max_int_steps,
+                                                 &found_flag, &uniform_generator, normal_rng_vec.data(),
+                                                 next_point_winner.data());
     printf(OL_ANSI_COLOR_CYAN "EI OPTIMIZATION FINISHED (wrong hyperparameters). Success status: %s\n" OL_ANSI_COLOR_RESET, found_flag ? "True" : "False");
     printf("Next best sample point according to EI (wrong hyper):\n");
     PrintMatrix(next_point_winner.data(), 1, dim);

--- a/moe/optimal_learning/cpp/gpp_hyperparameter_optimization_demo.cpp
+++ b/moe/optimal_learning/cpp/gpp_hyperparameter_optimization_demo.cpp
@@ -107,10 +107,12 @@ int main() {
   // hyperparameters later on
   // Generate hyperparameters randomly
   boost::uniform_real<double> uniform_double_for_hyperparameter(0.5, 1.5);
-  FillRandomCovarianceHyperparameters(uniform_double_for_hyperparameter, &uniform_generator, &hyperparameters_original, &covariance_original);
+  FillRandomCovarianceHyperparameters(uniform_double_for_hyperparameter, &uniform_generator,
+                                      &hyperparameters_original, &covariance_original);
 
   std::vector<ClosedInterval> hyperparameter_domain_bounds(covariance_original.GetNumberOfHyperparameters(), {1.0e-10, 1.0e10});
-  HyperparameterDomainType hyperparameter_domain(hyperparameter_domain_bounds.data(), covariance_original.GetNumberOfHyperparameters());
+  HyperparameterDomainType hyperparameter_domain(hyperparameter_domain_bounds.data(),
+                                                 covariance_original.GetNumberOfHyperparameters());
 
   // now fill data
 #if OL_USER_INPUTS == 1
@@ -134,9 +136,11 @@ int main() {
   domain.GenerateUniformPointsInDomain(num_sampled, &uniform_generator, points_sampled.data());
 
   // build an empty GP: since num_sampled (last arg) is 0, none of the data arrays will be used here
-  GaussianProcess gp_generator(covariance_original, points_sampled.data(), points_sampled_value.data(), noise_variance.data(), dim, 0);
+  GaussianProcess gp_generator(covariance_original, points_sampled.data(), points_sampled_value.data(),
+                               noise_variance.data(), dim, 0);
   // fill the GP with randomly generated data
-  FillRandomGaussianProcess(points_sampled.data(), noise_variance.data(), dim, num_sampled, points_sampled_value.data(), &gp_generator);
+  FillRandomGaussianProcess(points_sampled.data(), noise_variance.data(), dim, num_sampled,
+                            points_sampled_value.data(), &gp_generator);
 
   // choose a random initial guess reasonably far away from hyperparameters_original
   // to find some optima (see WARNING2 below), it may be necessary to start with hyperparameters smaller than the originals or
@@ -144,13 +148,15 @@ int main() {
   std::vector<double> hyperparameters_perturbed(covariance_original.GetNumberOfHyperparameters());
   CovarianceClass covariance_perturbed(dim, 1.0, 1.0);
   boost::uniform_real<double> uniform_double_for_wrong_hyperparameter(5.0, 12.0);
-  FillRandomCovarianceHyperparameters(uniform_double_for_wrong_hyperparameter, &uniform_generator, &hyperparameters_perturbed, &covariance_perturbed);
+  FillRandomCovarianceHyperparameters(uniform_double_for_wrong_hyperparameter, &uniform_generator,
+                                      &hyperparameters_perturbed, &covariance_perturbed);
 #endif
 
   // log likelihood type selection
   using LogLikelihoodEvaluator = LogMarginalLikelihoodEvaluator;
   // log likelihood evaluator object
-  LogLikelihoodEvaluator log_marginal_eval(points_sampled.data(), points_sampled_value.data(), noise_variance.data(), dim, num_sampled);
+  LogLikelihoodEvaluator log_marginal_eval(points_sampled.data(), points_sampled_value.data(),
+                                           noise_variance.data(), dim, num_sampled);
 
   int total_newton_errors = 0;  // number of newton runs that failed due to singular hessians
   int newton_max_num_steps = 500;  // max number of newton steps
@@ -158,7 +164,8 @@ int main() {
   double pre_mult_newton = 1.0e-1;  // newton diagonal dominance scaling factor (see newton docs for details)
   double max_relative_change_newton = 1.0;
   double tolerance_newton = 1.0e-11;
-  NewtonParameters newton_parameters(1, newton_max_num_steps, gamma_newton, pre_mult_newton, max_relative_change_newton, tolerance_newton);
+  NewtonParameters newton_parameters(1, newton_max_num_steps, gamma_newton, pre_mult_newton,
+                                     max_relative_change_newton, tolerance_newton);
 
   // call newton to optimize hyperparameters
   // in general if this takes the full hyperparameter_max_num_steps iterations, something went wrong
@@ -171,7 +178,9 @@ int main() {
 
   printf(OL_ANSI_COLOR_CYAN "NEWTON OPTIMIZED HYPERPARAMETERS:\n" OL_ANSI_COLOR_RESET);
   // run newton optimization
-  total_newton_errors += NewtonHyperparameterOptimization(log_marginal_eval, covariance_perturbed, newton_parameters, hyperparameter_domain, new_newton_hyperparameters.data());
+  total_newton_errors += NewtonHyperparameterOptimization(log_marginal_eval, covariance_perturbed,
+                                                          newton_parameters, hyperparameter_domain,
+                                                          new_newton_hyperparameters.data());
   // WARNING: the gradient of log marginal appears to go to 0 as you move toward infinity.  if you do not start
   // close enough to an optima or have overly aggressive diagonal dominance settings, newton will skip miss everything
   // going on locally and shoot out to these solutions.
@@ -203,23 +212,27 @@ int main() {
   printf(OL_ANSI_COLOR_CYAN "LOG LIKELIHOOD + GRADIENT AT NEWTON OPTIMIZED FINAL HYPERPARAMS:\n" OL_ANSI_COLOR_RESET);
 
   CovarianceClass covariance_final(dim, new_newton_hyperparameters[0], new_newton_hyperparameters.data() + 1);
-  typename LogLikelihoodEvaluator::StateType log_marginal_state_newton_optimized_hyper(log_marginal_eval, covariance_final);
+  typename LogLikelihoodEvaluator::StateType log_marginal_state_newton_optimized_hyper(log_marginal_eval,
+                                                                                       covariance_final);
   double newton_log_marginal_opt = log_marginal_eval.ComputeLogLikelihood(log_marginal_state_newton_optimized_hyper);
   printf("newton optimized log marginal likelihood = %.18E\n", newton_log_marginal_opt);
 
   std::vector<double> grad_log_marginal_opt(covariance_final.GetNumberOfHyperparameters());
-  log_marginal_eval.ComputeGradLogLikelihood(&log_marginal_state_newton_optimized_hyper, grad_log_marginal_opt.data());
+  log_marginal_eval.ComputeGradLogLikelihood(&log_marginal_state_newton_optimized_hyper,
+                                             grad_log_marginal_opt.data());
   printf("grad log likelihood: ");
   PrintMatrix(grad_log_marginal_opt.data(), 1, covariance_final.GetNumberOfHyperparameters());
 
   printf(OL_ANSI_COLOR_CYAN "LOG LIKELIHOOD + GRADIENT AT ORIGINAL HYPERPARAMS:\n" OL_ANSI_COLOR_RESET);
-  typename LogLikelihoodEvaluator::StateType log_marginal_state_original_hyper(log_marginal_eval, covariance_original);
+  typename LogLikelihoodEvaluator::StateType log_marginal_state_original_hyper(log_marginal_eval,
+                                                                               covariance_original);
 
   double original_log_marginal = log_marginal_eval.ComputeLogLikelihood(log_marginal_state_original_hyper);
   printf("original log marginal likelihood = %.18E\n", original_log_marginal);
 
   std::vector<double> original_grad_log_marginal(covariance_original.GetNumberOfHyperparameters());
-  log_marginal_eval.ComputeGradLogLikelihood(&log_marginal_state_original_hyper, original_grad_log_marginal.data());
+  log_marginal_eval.ComputeGradLogLikelihood(&log_marginal_state_original_hyper,
+                                             original_grad_log_marginal.data());
   printf("grad log likelihood: ");
   PrintMatrix(original_grad_log_marginal.data(), 1, covariance_original.GetNumberOfHyperparameters());
 

